### ISSUE TITLE
[Bug]: rule AL01 is not applied

### ISSUE THREAD

**Issue by bowmanjd** [quarylabs/sqruff#1131]
### What happened?

Using sqruff 0.21.12 on Linux, when I run

`echo 'select foo.bar from table1 foo' | sqruff fix -`

the result is unchanged:

```
select foo.bar from table1 foo
```

Even though we have explicit table aliasing on in the default config. What I would expect is what I get from sqlfluff:

```
echo 'select foo.bar from table1 foo' | sqlfluff fix --dialect ansi -
select foo.bar from table1 as foo  
```

It correctly adds the `as`.

As far as I can tell AL02 is working just fine (explicit column aliasing). It is just AL01 that is failing. The default config looks good for AL01, and I have even tried specifying a config with only:

```ini
[sqlfluff]
rules = AL01
```

and it still fails.

I am happy to help, and will keep looking into this.

### Version

0.21.12

### Checked in Playground

- [X] I have checked this bug in the [playground](https://playground.quary.dev/?secondary=Format).

**Comment by benfdking** [comment]
Hey @bowmanjd.

I'm sorry you've encountered this. I am investigating it at the moment, and it is odd that it seems to work in the playground. 

I've added a few tests to make sure we don't break things, and I am looking at fixing this!

**Comment by benfdking** [comment]
Ok, I went down a rabbit hole with other config problems that I found along the way ... I am looking your specific example now though and I can't seem to recreate it. If you could send me the full layout of the set up, I'd love to help as I have tried the following below.

In the [playground](https://playground.quary.dev/?secondary=Format) with config 

```
[sqruff]
dialect = ansi
rules = AL01
```

and SQL 

```sql 
select foo.bar test from table1 foo
```

it returns 

```
select foo.bar test from table1 AS foo
```

and so it seems to work. 

Locally with version `0.22.0` I can't seem to recreate the issue either with my config in the file `.sqruff`

My best guess is that `sqruff` is 
1. not picking up your config 
2. assuming `rules = core`
3. which does not include `AL01` and so there's no change in output.

**Comment by bowmanjd** [comment]
Yes, @benfdking I think your guess is right, too. I, too, can now see that the playground works fine (now that I am not overlooking the config options!)

However, something is still going wrong on my system, and I agree it must not be picking up the config. Here is my `/tmp/sqruff.conf` config now:

```ini
[sqruff]
dialect = ansi
rules = AL01
```

and here is the exact command I am using:
```sh
echo 'select foo.bar foobar from table1 foo' | target/debug/sqruff --config /tmp/sqruff.conf fix -
```

and here is the output:
```
== [<string>] FAIL
L:   1 | P:  16 | AL02 | Implicit/explicit aliasing of columns.
                       | [aliasing.column]
L:   1 | P:  16 | CP01 | Keywords must be consistently lower case.
                       | [capitalisation.keywords]
select foo.bar as foobar from table1 foo
```

which just doesn't make sense, does it? I have various `.sqlfluff` files lying around, but none are in a direct parent/ancestor of my current working directory.

Pretty sure this is something wrong on my side.

Wait... weird, weird! If I rename the file to `.sqruff` then it works!? This works fine:

```
echo 'select foo.bar foobar from table1 foo' | target/debug/sqruff --config /tmp/.sqruff fix -
```

Perhaps that is by design?

**Comment by benfdking** [comment]
Nope, it is not by design. The extension of the file doesn't matter. Have you pulled the latest version of the repo, I just made it so the `--config` now fails if the file is not found. Maybe it's not finding the file for some reason.

**Comment by bowmanjd** [comment]
Just a thought: could `load_config_at_path` check if the path is a valid file, and if so use that filename in place of `filename_options`?

**Comment by bowmanjd** [comment]
In other words, I think we have `filename_options` overriding the filename of the specified config, so that only `.sqruff` or `'.sqlfluff` are used.

**Comment by bowmanjd** [comment]
I think there are a variety of ways to solve it; let me know if a PR would be helpful.

**Comment by benfdking** [comment]
Always! Feel free to contribute, have you seen the test framework for the CLI? It's a bit repetitive but works and allows you to set up config files to show the error.
